os_net_config/utils.py

```diff
@@ -37,6 +37,7 @@
 
 [Service]
 Type=oneshot
+ExecStartPre=/usr/bin/udevadm settle
```

Comment:
There are still some possible edge cases where this would not work. The default timeout for "udevadm settle" is 120 seconds. We may want to increase this to something longer using --timeout=<seconds>, but we don't want to wait too long if the SR-IOV probing is done and udevadm is waiting for some other user rule to complete. If we know the sequence number of the rule we could use --seq-start=seqnum and --seq-stop=seqnum.

If we ran "udevadm settle" from within the os-net-config-sriov service could we do something like "udevadm settle --timeout=<seconds> --exit-if-exists=<file>" where the file is /sys/class/net/<ifname>/device/sriov_numvfs inside of get_numvfs in sriov_config.py in os-net-config when we call get_numvfs(pf_device['name']).

Reply:
I've pushed an update. Tested it briefly and it seems to do the trick.